clang/include/clang/Basic/DiagnosticDriverKinds.td

```diff
@@ -96,6 +96,15 @@ def err_drv_cuda_host_arch : Error<
 def err_drv_no_sycl_libspirv : Error<
   "cannot find '%0'; provide path to libspirv library via '-fsycl-libspirv-path', or "
   "pass '-fno-sycl-libspirv' to build without linking with libspirv">;
+def warn_flag_no_sycl_libspirv
```

Comment:
It's a little tricky. `-fno-sycl-libspirv` was introduced in case someone really does want to build dpc++ for cuda or hip without linking libspirv, so to say it is an invalid option is confusing I think. 
I think that option 2 is the best: It will deter people from using the flag inappropriately, and I think that if someone is using this flag for an unusual but correct reason they will not be thrown by the error message because they should understand that they will not get expected dpc++ behavior and that it is safe (for their purpose) to ignore the warning message.